htan_registry/tiffinfo.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import os
+import subprocess
+import re
+
+from synapsegenie.example_filetype_format import FileTypeFormat
+from synapsegenie import process_functions
+
+logger = logging.getLogger(__name__)
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+class TiffInfo(FileTypeFormat):
+
+    _filetype = "tiffinfo"
+
+    _process_kwargs = ["databaseSynId"]
+
+    def _validate_filetype(self, filePath):
+        assert os.path.basename(filePath[0]).endswith(".tif{1,2}|.svs")
+
+    def _process(self, df):
+        df.columns = [df.upper() for col in df.columns]
+        return df
+
+    def read_file(self, filePathList):
+        '''
+        Each file is to be read in for validation and processing.
+        This is not to be changed in any functions.
+
+        Args:
+            filePathList:  A list of file paths (Max is 2 for the two
+                           clinical files)
+
+        Returns:
+            df: Pandas dataframe of file
+        '''
+        # This file isn't a dataframe, so just return the filepath
+        filepath = filePathList[0]
+        return filepath
+
+    def process_steps(self, df, newPath, databaseSynId):
+        df = self._process(df)
+        process_functions.updateData(self.syn, databaseSynId, df, self.center,
+                                     toDelete=True)
+        df.to_csv(newPath, sep="\t", index=False)
+        return newPath
+
+    def _validate(self, path_or_df):
+        total_error = []
+        warning = []
+
+        # Validate the OME-XML with xmlvalid
+        xmlvalid_cmd = os.path.join(here, '../bftools/xmlvalid')
+        xmlvalid_exc = subprocess.run([xmlvalid_cmd, path_or_df], capture_output=True, text=True)
+        # If error, set the set a warning
+        if xmlvalid_exc.returncode:
+            warning.append("Invalid OME-XML")
+
+        # Get all information on the file from tiffdump
+        tiffdump_exc = subprocess.run(["tiffinfo", path_or_df], capture_output=True, text=True)
```

Comment:
What is `tiffinfo`?  Is it part of `bftools`?

Reply:
It is part of `libtiff-tools` and can be used to print out information from the headers
> Tiffinfo displays information about files created according to the Tag Image File Format, Revision 6.0. By default, the contents of each TIFF directory in each file is displayed, with the value of each tag shown symbolically (where sensible).

Install with `sudo apt install libtiff-tools` or equivalent

Reply:
We can add this to requirements or documentation